### ISSUE TITLE
StEP: Style-based Encoder Pre-training for Multi-modal Image Synthesis

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#119]
*CVPR 2021*

# TL;DR

- Pretrain a ***style encoder*** by ***Triplet loss*** (positive and negative samples designated by pretrained VGG style similarity)
- Train generator *(with style encoder fixed)* on LS-GAN and perceptual reconstruction loss, using the encoded target style.

# Motivation

- Multi-modal image-to-image translation requires understanding of rich latent space distribution.
- Learn rich latent space while maintaining simple training objective: hence more efficient training and robustness to hyper-parameters.

# Previous Work

- BicycleGAN: enforce one-to-one mapping between a random noise and translated image so that the generator ***cannot ignore*** the latent code.
- MUNIT

# Method

![image](https://user-images.githubusercontent.com/46308407/131272909-eb0270b8-9fb0-4970-af78-1a84d75b606f.png)

## Stage 1
- Given a query image, use pretrained VGG to compute style similarities with other images in the dataset and construct k-nearest and k-farthest group.
- Note that k-farthest group is larger than k-nearest group, since few style outliers can dominate negative samples in most of the cases.
- Sample one each from nearest and farthest group, and train encoder on triplet loss.
![image](https://user-images.githubusercontent.com/46308407/131272999-79dece31-0794-493e-a76b-754715d74fe4.png)
- Thus, encoder learns to *cluster* images based on their styles. (SSL approach)

## Stage 2
-  Now that we have an encoded style for the target image, we fix the style encoder and train generator on GAN loss + perceptual reconstruction loss.
- Least Squares GAN loss is used for adversarial supervision.
![image](https://user-images.githubusercontent.com/46308407/131273079-6c408ee0-5863-4182-ad37-6d31438d91d0.png)

## Stage 3
- Finetune both style encoder and the generator simultaneously.

## Style Sampling
- Style of an existing image can be simply obtained by feeding the image to the style encoder.
- For direct sampling from the latent distribution, train an ***auxiliary mapping network*** that maps a unit Gaussian noise to the learned latent space.
- Mapping network is learned as a postprocessing step using implicit MLE.
![image](https://user-images.githubusercontent.com/46308407/131273217-72bb9adc-51f6-4e1d-863c-7f885cbda88a.png)

# Experiments

## Datasets
- Architectural Labels-Photo
- Aerial-Map
- Edges-Shoes/Handbags
- Night-Day
- Space Needle Timelapse

## Metric
- PSNR, Alex-net LPIPS

## Baseline
![image](https://user-images.githubusercontent.com/46308407/131274043-1e6ed632-e8cc-4db1-9e51-bc27c47345c1.png)
![image](https://user-images.githubusercontent.com/46308407/131274055-750d6254-1de6-4a2c-aad6-bd0ce2cb2ae7.png)
![image](https://user-images.githubusercontent.com/46308407/131274190-e0726e3c-9b81-4442-8fac-dca673639adb.png)

## Analysis
![image](https://user-images.githubusercontent.com/46308407/131274091-47117dd7-dca1-4a10-8658-c9fb8e6eeb3f.png)
![image](https://user-images.githubusercontent.com/46308407/131274099-f0d3f42f-8d24-4915-ac87-53d36a0c5501.png)
![image](https://user-images.githubusercontent.com/46308407/131274110-59f19648-b6cd-4a16-94d1-4e5986ccc4b5.png)

- Learned style latent space is ***generalizable***. That is, style encoder can learn style from one dataset and that can be applied to other domain.

![image](https://user-images.githubusercontent.com/46308407/131274130-5438ac57-f22f-4905-86f9-6952fbc0d3c1.png)

## Ablation
![image](https://user-images.githubusercontent.com/46308407/131274207-df8787d7-4bff-4cf8-aa89-2c8c7d60621a.png)

# Comments
- Simple yet effective.
- Ideas from SSL is employed (triplet loss)
- Simple loss function (similar to pix2pix)
- Multi-modal image translation via inputting different style codes.
